wait-for-usertask.js

```diff
@@ -0,0 +1,91 @@
+module.exports = function (RED) {
+    function WaitForUsertask(config) {
+        RED.nodes.createNode(this, config);
+        var node = this;
+
+        node.engine = RED.nodes.getNode(config.engine);
+
+        const client = node.engine.engineClient;
+
+        let subscription = null;
+        let currentIdentity = node.engine.identity;
+        let subscribe = null;
+
+        node.on('input', async function (msg) {
+            subscribe = async () => {
+                if (!client) {
+                    node.error('No engine configured.');
+                    return;
+                }
+
+                const query = RED.util.evaluateNodeProperty(config.query, config.query_type, node, msg);
+
+                subscription = await client.userTasks.onUserTaskWaiting(async (userTaskWaitingNotification) => {
+
+                    const newQuery = {
+                        'flowNodeInstanceId': userTaskWaitingNotification.flowNodeInstanceId,
+                        ...query
+                    };
+
+                    const matchingFlowNodes = await client.userTasks.query(newQuery, { identity: currentIdentity });
+
+                    if (matchingFlowNodes.userTasks && matchingFlowNodes.userTasks.length == 1) {
+                        const userTask = matchingFlowNodes.userTasks[0];
+
+                        msg.payload = { userTask: userTask };
+                        node.send(msg);
+                    } else {
+                        // nothing todo - wait for next notification
+                    }
+
+                    // remove subscription
+                    client.userTasks.removeSubscription(subscription, currentIdentity);
+                }, { identity: currentIdentity });
+
+                node.log({"Handling old userTasks config.only_for_new": config.only_for_new});
+
+                if (config.only_for_new === false) {
+                    // only check suspended user tasks
+                    const suspendedQuery = {
+                        'state': 'suspended',
+                        ...query
+                    };
+
+                    const matchingFlowNodes = await client.userTasks.query(suspendedQuery, { identity: currentIdentity });
+
+                    if (matchingFlowNodes.userTasks && matchingFlowNodes.userTasks.length >= 1) {
+                        const userTask = matchingFlowNodes.userTasks[0];
+
+                        msg.payload = { userTask: userTask };
+                        node.send(msg);
+
+                        // remove subscription
+                        client.userTasks.removeSubscription(subscription, currentIdentity);
+                    } else {
+                        // let the *currentIdentity* be active
+                    }
+                }
```

Comment:
Der Code müsste eigentlich passieren bevor die Subscription erstellt wird, oder nicht? Sonst könnte es ja sein, dass eine Subscription eine Notification erhält und den Task zurück gibt, obwohl only for new auf false gestellt wurde. 

Also müsste der Code über das Erstellen der Subscription und die Subscription nur erstellt werden, wenn only for new auf true gestellt ist. Im Code muss dann natürlich Zeile 63 nicht mehr gemacht werden.